### ISSUE TITLE
Webpack5 ESM strict mode

### ISSUE THREAD

**Issue by ocdi** [babel/babel#12058]
## Bug Report

Webpack5 `beta.30` has enabled strict mode on the ESM modules imports https://github.com/webpack/webpack/releases/tag/v5.0.0-beta.30

https://github.com/webpack/webpack/issues/11467 has noted people with code problems, however `@babel/runtime` appears to have code that doesn't follow the strict `"type": "module"` requirements, as expected for node / webpack.

`.\node_modules\@babel\runtime\helpers\esm\package.json` has `"type": "module"` being set and libraries that depend on `@babel/runtime` with ESM modules are broken as a result.

I expect the code generator should be outputting with the file extensions to satisfy the requirements of ESM modules

https://nodejs.org/api/esm.html#esm_differences_between_es_modules_and_commonjs

**Current behavior**

@babel/runtime/helpers/esm/*.js contains extensionless imports for dependencies.

**Expected behavior**

@babel/runtime/helpers/esm/*.js to contain fully specified extensions for imports

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**

N/A - this is npm published module

**Environment**

```
  System:
    OS: Windows 10 10.0.19042
  Binaries:
    Node: 12.18.3 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.8 - C:\s\SwiftFox\node_modules\.bin\npm.CMD
  Monorepos:
    Yarn Workspaces: 1.22.5
  npmPackages:
    @babel/core: 7.11.4 => 7.11.4
    babel-loader: 8.1.0 => 8.1.0
    eslint: 7.7.0 => 7.7.0
    webpack: 5.0.0-beta.30 => 5.0.0-beta.30
```

**Comment by sokra** [comment]
btw this is not an webpack-only error. If you run `import("@babel/runtime-corejs3/helpers/esm/defineProperty.js")` in node.js you get an error too:

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '...\node_modules\@babel\runtime-corejs3\core-js\object\define-property' imported from ...\node_modules\@babel\runtime-corejs3\helpers\esm\defineProperty.js
```

**Comment by shpyo** [comment]
True, this is causing my issue:
```

14-Sep-2020 10:46:41 | ERROR in ./node_modules/@babel/runtime/helpers/esm/inherits.js 1:0-46
-- | --
14-Sep-2020 10:46:41 | Module not found: Error: Can't resolve './setPrototypeOf' in '/code/node_modules/@babel/runtime/helpers/esm'
```

**Comment by JLHwung** [comment]
Duplicate of #8462

**Comment by JLHwung** [comment]
This issue will be fixed by #10853, please leave a review comment there, thanks.

**Comment by evilebottnawi** [comment]
@JLHwung Can we prioritize the problem?

**Comment by JLHwung** [comment]
@evilebottnawi I will ship it asap once that PR is approved. Can you leave a review?

**Comment by felixfbecker** [comment]
Webpack 5 is now released so this would be good to release asap as many people will run into it now

**Comment by manuelbieh** [comment]
Indeed. Came here via Google. Ran into it right away. Guess I won't remain the only one. Affected modules:
```
./setPrototypeOf
./defineProperty
../../helpers/esm/typeof
./assertThisInitialized
```
Any hints on how I can solve this? I see a merged PR from 26 days ago but it was for Next.js, not for Babel.

**Comment by nicolo-ribaudo** [comment]
We plan to release 7.12, which fixes this issue, the upcoming week.

**Comment by arpu** [comment]
same problem here any workaround available before Release?

**Comment by mustafawm** [comment]
@arpu the package mentioned here might help https://github.com/babel/babel/issues/8462#issuecomment-609651837

**Comment by piecyk** [comment]
@arpu till then you can add rule for resolving those modules 

https://github.com/webpack/webpack/issues/11467#issuecomment-691873586

**Comment by SokichiFujita** [comment]
babel 7.12.0 resolved the errors with webpack v5. nice.
